### PR TITLE
Allow parsing decimals from strings

### LINKED ISSUE #342: [JIT] dbsp.date.epoch should return i64
Using 32 bits overflows the result in 2038.

### PR DESCRIPTION
**Pull request by Kixiron**
Fixes #342 and #340

### CODE REVIEW

File: crates/dataflow-jit/src/codegen/intrinsics.rs [hunk 1]
```diff
@@ -1336,6 +1337,24 @@ unsafe extern "C" fn parse_bool_from_str(
     }
 }
 
+unsafe extern "C" fn parse_decimal_from_str(
+    ptr: *const u8,
+    len: usize,
+    output: &mut MaybeUninit<u128>,
+) -> bool {
+    let string = unsafe { str_from_raw_parts(ptr, len) };
+    match string.parse::<Decimal>() {
```
Comment (mihaibudiu) [inline crates/dataflow-jit/src/codegen/intrinsics.rs:1346]: https://docs.feldera.io/docs/sql/decimal: Casting a string to a decimal value will produce the value 0 when parsing fails.
In general this behavior depends on the actual data type, sometimes is should be NULL.

Comment (Kixiron) [reply crates/dataflow-jit/src/codegen/intrinsics.rs:1346]: Yah, in the future I'll add the ability to react to parse failures but it depends on adts

Comment (mihaibudiu) [reply crates/dataflow-jit/src/codegen/intrinsics.rs:1346]: Until then I will file an issue about this after we merge this PR.

Comment (Kixiron) [reply crates/dataflow-jit/src/codegen/intrinsics.rs:1346]: I believe there's already one regarding this

**Pull request merged**